substrate/frame/support/procedural/src/pallet/expand/event.rs

```diff
@@ -102,25 +102,37 @@ pub fn expand_event(def: &mut Def) -> proc_macro2::TokenStream {
 	let default_deprecation_info =
 		quote::quote! { #frame_support::__private::metadata_ir::DeprecationStatus::NotDeprecated }
 			.to_string();
-
-	let variants: Vec<proc_macro2::TokenStream> = event_item
-		.variants
-		.iter()
-		.enumerate()
-		.filter_map(|(index, x)| {
-			let key = index as u8;
-			let deprecation_status =
-				crate::deprecation::get_deprecation(&quote::quote! {#frame_support}, &x.attrs)
+	let deprecation = {
```

Comment:
Could we move this bit into the `deprecation` module? Such that we share the code between calls, errors and events?

Reply:
refactored this, now the code should be shared